### ISSUE TITLE
Client / local cluster not working

### ISSUE THREAD

**Issue by N4321D** [dask/dask#10445]
My scripts stopped working after upgrading to 2023.8.0 with python 3.11. 
I tracked it down to creating a client and localcluster. 
If I disable that everything works fine. 

## Example ##
```
import dask
import dask.dataframe as dd
import pandas as pd
from dask.distributed import Client

client = Client()  # script works if disabling this line

dd.from_pandas(pd.DataFrame({'test': [i for i in range(int(100))]}), npartitions=1).mean().compute()
```

produces this error:

```
2023-08-16 13:30:23,485 - distributed.protocol.core - CRITICAL - Failed to deserialize
Traceback (most recent call last):
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/protocol/core.py", line 158, in loads
    return msgpack.loads(
           ^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/msgpack/fallback.py", line 136, in unpackb
    raise ExtraData(ret, unpacker._get_extradata())
msgpack.exceptions.ExtraData: unpack(b) received extra data.
2023-08-16 13:30:23,487 - distributed.core - ERROR - Exception while handling op register-client
Traceback (most recent call last):
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/core.py", line 930, in _handle_comm
    result = await result
             ^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/scheduler.py", line 5453, in add_client
    await self.handle_stream(comm=comm, extra={"client": client})
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/core.py", line 983, in handle_stream
    msgs = await comm.read()
           ^^^^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/comm/tcp.py", line 253, in read
    msg = await from_frames(
          ^^^^^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/comm/utils.py", line 100, in from_frames
    res = _from_frames()
          ^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/comm/utils.py", line 83, in _from_frames
    return protocol.loads(
           ^^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/protocol/core.py", line 158, in loads
    return msgpack.loads(
           ^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/msgpack/fallback.py", line 136, in unpackb
    raise ExtraData(ret, unpacker._get_extradata())
msgpack.exceptions.ExtraData: unpack(b) received extra data.
Task exception was never retrieved
future: <Task finished name='Task-204' coro=<Server._handle_comm() done, defined at /home/dmitri/miniconda3/envs/data/lib/python3.11/site-packages/distributed/core.py:836> exception=ExtraData(({'op': 'update-graph', 'graph_header': {'serializer': 'pickle', 'writeable': (True,)}, 'graph_frames': (b'\x80\x05\x95\x07\x0c\x00\x00\x00\x00\x00\x00\x8c\x1edistributed.protocol.serialize\x94\x8c\x08ToPickle\x94\x93\x94)\x81\x94}\x94\x8c\x04data\x94\x8c\x13dask.highlevelgraph\x94\x8c\x0eHighLevelGraph\x94\x93\x94)\x81\x94}\x94(\x8c\x0cdependencies\x94}\x94(\x8cPdataframe-mean-7e3bb5d64a08036e23fcbf93a9bab08b-99da06c12cac5b1fa074bf4e015a297d\x94\x8f\x94(\x8c4dataframe-count-agg-1f8ba43dc756a946115daec94fb7e381\x94\x8c2dataframe-sum-agg-9f1fdc1a76ed99888426a07407e7a73c\x94\x90h\x10\x8f\x94(\x8cUdataframe-sum-chunk-9f1fdc1a76ed99888426a07407e7a73c-8ce7a99b459fe91ae20b5e788f8d39d2\x94\x90h\x12\x8f\x94(\x8c,from_pandas-1df5853bd65ae179c1c10d82e9be470a\x94\x90h\x0f\x8f\x94(\x8cWdataframe-count-chunk-1f8ba43dc756a946115daec94fb7e381-7b814e550a86f49aea6ab00405eec468\x94\x90h\x16\x8f\x94(h\x14\x90h\x14\x8f\x94u\x8c\x10key_dependencies\x94}\x94\x8c\x06layers\x94}\x94(h\r\x8c\x0edask.blockwise\x94\x8c\tBlockwise\x94\x93\x94)\x81\x94}\x94(\x8c\x0bannotations\x94N\x8c\x16collection_annotations\x94N\x8c\x06output\x94h\r\x8c\x0eoutput_indices\x94\x8c\x02.0\x94\x85\x94\x8c\routput_blocks\x94N\x8c\x03dsk\x94}\x94h\r\x8c\x16dask.dataframe.methods\x94\x8c\x0emean_aggregate\x94\x93\x94\x8c\x13__dask_blockwise__0\x94\x8c\x13__dask_blockwise__1\x94\x87\x94s\x8c\tnumblocks\x94}\x94(h\x10K\x01\x85\x94h\x0fK\x01\x85\x94u\x8c\x07io_deps\x94}\x94\x8c\x07indices\x94h\x10h&\x85\x94\x86\x94h\x0fh&\x85\x94\x86\x94\x86\x94\x8c\x0bconcatenate\x94N\x8c\x08new_axes\x94}\x94\x8c\x05_dims\x94}\x94h&K\x01subh\x10\x8c\x0bdask.layers\x94\x8c\x16DataFrameTreeReduction\x94\x93\x94)\x81\x94}\x94(h"Nh#N\x8c\x04name\x94h\x10\x8c\nname_input\x94h\x12\x8c\x11npartitions_input\x94K\x01\x8c\x0bconcat_func\x94\x8c\tfunctools\x94\x8c\x07partial\x94\x93\x94\x8c\x13dask.dataframe.core\x94\x8c\x07_concat\x94\x93\x94\x85\x94R\x94(hP)}\x94\x8c\x0cignore_index\x94\x89sNt\x94b\x8c\x0etree_node_func\x94hMhN\x8c\x12_reduction_combine\x94\x93\x94\x85\x94R\x94(hX)}\x94(\x8c\x0baca_combine\x94hN\x8c\x15_getattr_numeric_only\x94\x93\x94\x8c\x06skipna\x94\x88\x8c\x04axis\x94K\x00\x8c\x11_dask_method_name\x94\x8c\x03sum\x94\x8c\x0cnumeric_only\x94NuNt\x94b\x8c\rfinalize_func\x94hMhN\x8c\x14_reduction_aggregate\x94\x93\x94\x85\x94R\x94(hg)}\x94(\x8c\raca_aggregate\x94h^h_\x88h`K\x00hahbhcNuNt\x94b\x8c\x0bsplit_every\x94K\x01\x8c\tsplit_out\x94N\x8c\x11output_partitions\x94]\x94K\x00a\x8c\x0etree_node_name\x94\x8c6dataframe-sum-combine-9f1fdc1a76ed99888426a07407e7a73c\x94\x8c\x06widths\x94]\x94K\x01a\x8c\x06height\x94K\x01\x8c\x13_cached_output_keys\x94\x8f\x94(h\x10K\x00\x86\x94\x90ubh\x12h\x1f)\x81\x94}\x94(h"Nh#Nh$h\x12h%\x8c\x02.0\x94\x85\x94h(Nh)}\x94h\x12(\x8c\ndask.utils\x94\x8c\x05apply\x94\x93\x94hN\x8c\x10_reduction_chunk\x94\x93\x94]\x94\x8c\x13__dask_blockwise__0\x94a\x8c\x08builtins\x94\x8c\x04dict\x94\x93\x94]\x94(]\x94(\x8c\taca_chunk\x94h^e]\x94(h_\x88e]\x94(h`K\x00e]\x94(hahbe]\x94(hcNee\x86\x94t\x94sh1}\x94h\x14K\x01\x85\x94sh5}\x94h7h\x14h{\x85\x94\x86\x94\x85\x94h=Nh>}\x94h@}\x94h{K\x01subh\x0fhD)\x81\x94}\x94(h"Nh#NhGh\x0fhHh\x16hIK\x01hJhMhP\x85\x94R\x94(hP)}\x94hT\x89sNt\x94bhVhMhX\x85\x94R\x94(hX)}\x94h\\hN\x8c\x10_count_aggregate\x94\x93\x94sNt\x94bhehMhg\x85\x94R\x94(hg)}\x94hkh\xa3sNt\x94bhmK\x01hnNho]\x94K\x00ahq\x8c8dataframe-count-combine-1f8ba43dc756a946115daec94fb7e381\x94hs]\x94K\x01ahuK\x01hv\x8f\x94(h\x0fK\x00\x86\x94\x90ubh\x16h\x1f)\x81\x94}\x94(h"Nh#Nh$h\x16h%\x8c\x02.0\x94\x85\x94h(Nh)}\x94h\x16(h\x80h\x82]\x94\x8c\x13__dask_blockwise__0\x94ah\x87]\x94(]\x94(hc\x89e]\x94(h\x8ah~\x8c\x0cmethodcaller\x94\x93\x94\x8c\x05count\x94\x85\x94R\x94ee\x86\x94t\x94sh1}\x94h\x14K\x01\x85\x94sh5}\x94h7h\x14h\xb0\x85\x94\x86\x94\x85\x94h=Nh>}\x94h@}\x94h\xb0K\x01subh\x14h\x06\x8c\x11MaterializedLayer\x94\x93\x94)\x81\x94}\x94(h"Nh#}\x94(\x8c\x0bnpartitions\x94K\x01\x8c\x07columns\x94]\x94\x8c\x04test\x94a\x8c\x04type\x94\x8c\x1ddask.dataframe.core.DataFrame\x94\x8c\x0edataframe_type\x94\x8c\x1bpandas.core.frame.DataFrame\x94\x8c\rseries_dtypes\x94}\x94h\xcf\x8c\x05numpy\x94\x8c\x05dtype\x94\x93\x94\x8c\x02i8\x94\x89\x88\x87\x94R\x94(K\x03\x8c\x01<\x94NNNJ\xff\xff\xff\xffJ\xff\xff\xff\xffK\x00t\x94bsu\x8c\x07mapping\x94}\x94h\x14K\x00\x86\x94\x8c\x11pandas.core.frame\x94\x8c\tDataFrame\x94\x93\x94)\x81\x94}\x94(\x8c\x04_mgr\x94\x8c\x1epandas.core.internals.managers\x94\x8c\x0cBlockManager\x94\x93\x94\x8c\x16pandas._libs.internals\x94\x8c\x0f_unpickle_block\x94\x93\x94\x8c\x12numpy.core.numeric\x94\x8c\x0b_frombuffer\x94\x93\x94(\x97h\xdbK\x01K\n\x86\x94\x8c\x01C\x94t\x94R\x94h\x85\x8c\x05slice\x94\x93\x94K\x00K\x01K\x01\x87\x94R\x94K\x02\x87\x94R\x94\x85\x94]\x94(\x8c\x18pandas.core.indexes.base\x94\x8c\n_new_Index\x94\x93\x94h\xfc\x8c\x05Index\x94\x93\x94}\x94(h\x05\x8c\x15numpy.core.multiarray\x94\x8c\x0c_reconstruct\x94\x93\x94h\xd6\x8c\x07ndarray\x94\x93\x94K\x00\x85\x94C\x01b\x94\x87\x94R\x94(K\x01K\x01\x85\x94h\xd8\x8c\x02O8\x94\x89\x88\x87\x94R\x94(K\x03\x8c\x01|\x94NNNJ\xff\xff\xff\xffJ\xff\xff\xff\xffK?t\x94b\x89]\x94h\xcfat\x94bhGNu\x86\x94R\x94h\xfe\x8c\x19pandas.core.indexes.range\x94\x8c\nRangeIndex\x94\x93\x94}\x94(hGN\x8c\x05start\x94K\x00\x8c\x04stop\x94K\n\x8c\x04step\x94K\x01u\x86\x94R\x94e\x86\x94R\x94\x8c\x04_typ\x94\x8c\tdataframe\x94\x8c\t_metadata\x94]\x94\x8c\x05attrs\x94}\x94\x8c\x06_flags\x94}\x94\x8c\x17allows_duplicate_labels\x94\x88subsubuubsb.', b'\x00\x00\x00\x00\x00\x00\x00\x00'), 1: 0, 0: 0, 2: 0},), bytearray(b"\x00\x00\x03\x00\x00\x00\x00\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\x05\x00\x00\x00\x00\x00\x00\x00\x06\x00\x00\x00\x00\x00\x00\x00\x07\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x00\x00\x00\x00\t\x00\x00\x00\x00\x00\x00\x00\xa4keys\x91\xd9W(\'dataframe-mean-7e3bb5d64a08036e23fcbf93a9bab08b-99da06c12cac5b1fa074bf4e015a297d\', 0)\xb1internal_priority\xc0\xafsubmitting_task\xc0\xacfifo_timeout\xa360s\xa6actors\xc0\xa4code\x81\xab__Pickled__\x01\xabannotations\x81\xab__Pickled__\x02"))>
Traceback (most recent call last):
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/core.py", line 930, in _handle_comm
    result = await result
             ^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/scheduler.py", line 5453, in add_client
    await self.handle_stream(comm=comm, extra={"client": client})
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/core.py", line 983, in handle_stream
    msgs = await comm.read()
           ^^^^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/comm/tcp.py", line 253, in read
    msg = await from_frames(
          ^^^^^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/comm/utils.py", line 100, in from_frames
    res = _from_frames()
          ^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/comm/utils.py", line 83, in _from_frames
    return protocol.loads(
           ^^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/distributed/protocol/core.py", line 158, in loads
    return msgpack.loads(
           ^^^^^^^^^^^^^^
  File "/home/d/miniconda3/envs/data/lib/python3.11/site-packages/msgpack/fallback.py", line 136, in unpackb
    raise ExtraData(ret, unpacker._get_extradata())
msgpack.exceptions.ExtraData: unpack(b) received extra data.
---------------------------------------------------------------------------
CancelledError                            Traceback (most recent call last)
Cell In[5], line 1
----> 1 dd.from_pandas(pd.DataFrame({'test': [i for i in range(int(10))]}), npartitions=1).mean().compute()

File ~/miniconda3/envs/data/lib/python3.11/site-packages/distributed/client.py:2268, in Client._gather(self, futures, errors, direct, local_worker)
   2266     else:
   2267         raise exception.with_traceback(traceback)
-> 2268     raise exc
   2269 if errors == "skip":
   2270     bad_keys.add(key)

CancelledError: ('dataframe-mean-7e3bb5d64a08036e23fcbf93a9bab08b-99da06c12cac5b1fa074bf4e015a297d', 0)
```

**Anything else we need to know?**:
The script works in python 3.9 + dask 2023.6.0 env

**Environment**:

- Dask version: 2023.8.0
- Distributed version: 2023.8.0
- Python version: 3.11.4
- Operating System: Ubuntu 22.04.3 LTS
- Install method (conda, pip, source): Conda (conda-forge)

**Comment by jrbourbeau** [comment]
Thanks for the issue @N4321D. What version of `msgpack` are you using? It looks like there was a similar issue reported that's been fixed in the latest `msgpack=1.0.5` release (xref https://github.com/msgpack/msgpack-python/issues/526)

**Comment by N4321D** [comment]
Thank you @jrbourbeau

i was using msgpack-python            1.0.3 

upgraded to 1.0.5 and that fixed it! 

Thanks again for the help!

**Comment by N4321D** [comment]
fixed by upgrading msgpack-python to 1.0.5
```conda install -c conda-forge msgpack-python=1.0.5```

**Comment by jrbourbeau** [comment]
Glad to hear it
